lib/components/CircuitJsonPreview/CircuitJsonPreview.tsx

```diff
@@ -121,6 +121,18 @@ export const CircuitJsonPreview = ({
     setIsFullScreen(!isFullScreen)
   }
 
+  useEffect(() => {
+    if (isFullScreen) {
+      document.body.style.overflow = "hidden"
+    } else {
+      document.body.style.overflow = ""
+    }
+
+    return () => {
+      document.body.style.overflow = ""
+    }
+  }, [isFullScreen])
```

Comment:
can you isolate this hook to a separate file? These components tend to grow out of control in size with little hacks